src/me.rs

```diff
@@ -39,6 +38,79 @@ pub fn get_sad(
   sum
 }
 
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+#[target_feature(enable = "avx2")]
+unsafe fn get_sad_avx2(
+  plane_org: &mut PlaneSlice, plane_ref: &mut PlaneSlice, blk_h: usize,
+  blk_w: usize
+) -> u32 {
+  #[cfg(target_arch = "x86")]
+  use std::arch::x86::*;
+  #[cfg(target_arch = "x86_64")]
+  use std::arch::x86_64::*;
+
+  let mut sum = 0 as u32;
+  let size;
+
+  {
+    let slice_org = plane_org.as_slice_w_width(blk_w);
+    let slice_ref = plane_ref.as_slice_w_width(blk_w);
+    size = ::std::cmp::min(slice_org.len(), slice_ref.len()) as isize;
+
+    let mut ptr_org = slice_org.as_ptr();
+    let mut ptr_ref = slice_ref.as_ptr();
+
+    let mut v32x8sum = _mm256_set1_epi8(0);
+
+    for _r in 0..blk_h {
+      let mut pos = 0;
+      while pos <= size-8 {
+        let v16x8org = _mm_load_si128(ptr_org.offset(pos) as *const _);
+        let v16x8ref = _mm_load_si128(ptr_ref.offset(pos) as *const _);
+        let v32x8org = _mm256_cvtepu16_epi32(v16x8org);
+        let v32x8ref = _mm256_cvtepu16_epi32(v16x8ref);
+        let v32x8diff = _mm256_sub_epi32(v32x8org, v32x8ref);
```

Comment:
org and ref will contain either 12/10/8-bit values so it's safe to subtract and abs before converting to 32 bits. If you are working with 128-bit vectors, it's best to use madd to convert to 32-bits and add horizontally. That being said, it's faster to add a few lines as 16 bits.